src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -2028,14 +2024,12 @@ BasicType SystemDictionary::box_klass_type(Klass* k) {
 
 #ifdef ASSERT
 // Verify that this placeholder exists since this class is in the middle of loading.
-void verify_placeholder(Symbol* class_name, ClassLoaderData* loader_data, bool is_parallel_capable) {
-  // Only parallel capable class loaders use placeholder table for define class (bizarre)
-  if (is_parallel_capable) {
-    assert_locked_or_safepoint(SystemDictionary_lock);
-    unsigned int name_hash = placeholders()->compute_hash(class_name);
-    Symbol* ph_check =  placeholders()->find_entry(name_hash, class_name, loader_data);
-    assert(ph_check != NULL, "This placeholder should exist");
-  }
+void verify_placeholder(Symbol* class_name, ClassLoaderData* loader_data) {
+  // Only parallel capable class loaders use placeholder table for define class.
```

Comment:
Maybe I'm having trouble following the call sequence, but non-parallel capable loaders also add placeholder entries, and then call check_constraints. So why do we not need to verify the placeholder in that case?

Reply:
Non-parallel capable class loaders do NOT add placeholder entries for DEFINE_CLASS (capitalized because that's the enum).

Reply:
Sorry I'm still not seeing why we only verify DEFINE_CLASS placeholder entries?

Reply:
Actually, we call verify_constraints (where this verify_placeholder call is done) twice.  Once it verifies that there's a LOAD_INSTANCE placeholder from resolve_instance_class_or_null, and one path verifies there's a DEFINE_CLASS placeholder which is only used for bootstrap and parallelCapable.  The verify_placeholder function only verifies that there's a placeholder present, not which one. The code that was there was essentially a nop so I replaced it with something that does something in this change, but I think I want to eventually delete it.